src/main/java/uk/gov/register/presentation/resource/SearchResource.java

```diff
@@ -42,7 +42,7 @@ public Response findByPrimaryKey(@PathParam("primaryKey") String key, @PathParam
             return buildResponse(new SingleResultView("/templates/entry.mustache", entry.isPresent() ? entry.get() : null));
         }
 
-        throw new BadRequestException("Key: " + key + " is not primary key of the register.");
+        throw new NotFoundException();
```

Comment:
In my opinion it should be 400. For a particular register we don't allow search with any key other than primary key. If you try to search it is bad request because you typed wrong key. while 404 in this particular case means that the 'key' is correct but there is no resource found with the value.